tests/lib/devicetree/api/src/main.c

```diff
@@ -1625,6 +1625,17 @@ static void test_path(void)
 			     "/test/gpio@deadbeef"), "");
 }
 
+static void test_node_name(void)
+{
+	zassert_true(!strcmp(DT_NODE_FULL_NAME(DT_ROOT), "/"), "");
+	zassert_true(!strcmp(DT_NODE_FULL_NAME(TEST_DEADBEEF),
+			     "gpio@deadbeef"), "");
+	zassert_true(!strcmp(DT_NODE_FULL_NAME(TEST_TEMP),
+			     "temperature-sensor"), "");
+	zassert_true(strcmp(DT_NODE_FULL_NAME(TEST_REG),
+			     "reg-holder"), "");
```

Comment:
Why isn't this `"reg-holder@9999aaaa"`?

Reply:
Because I'm testing that it does NOT match.

Reply:
Ah.  A little subtle.